src/main/java/com/github/pedrovgs/androidwifiadb/adb/AndroidWiFiADB.java

```diff
@@ -0,0 +1,161 @@
+   * Call <code>getDevices()</code> in order to get updated devices list.
+   * <br>We should always update devices list,
+   * because their names might change.
+   * Unathorized and authorized states produce different device names.
+   * @return true - refresh required, false - otherwise.
+   */
+  public boolean refreshDevicesList() {
+    if (!isADBInstalled()) {
+      return false;
+    }
+    List<Device> connected = adb.getDevicesConnectedByUSB();
+    for (Device connectedDevice : connected) {
+      boolean deviceExists = checkDeviceExistance(connectedDevice);
+      if (!deviceExists) {
+        connectedDevice.setIp(adb.getDeviceIp(connectedDevice));
+        devices.add(connectedDevice);
+      } else {
+        updateDeviceInfo(connectedDevice);
+      }
+    }
+    return true;
+  }
+
+  private void updateDeviceInfo(Device updatedDevice) {
+    for (int i = 0, size = devices.size(); i < size; i++) {
+      Device device = devices.get(i);
+      if (updatedDevice.getId().equals(device.getId())) {
+        devices.remove(i);
+        device.setName(updatedDevice.getName());
+        devices.add(i, device);
+      }
+    }
+  }
+
+  private boolean checkDeviceExistance(Device connectedDevice) {
+    boolean deviceExists = false;
+    for (Device device : devices) {
+      if (connectedDevice.getId().equals(device.getId())) {
+        deviceExists = true;
+      }
+    }
+    return deviceExists;
+  }
+
+  public List<Device> getDevices() {
+    return devices;
+  }
+
+  private boolean isADBInstalled() {
+    return adb.isInstalled();
+  }
+
+  public void updateProject(Project project) {
+    this.adb.updateProject(project);
+  }
+
+  public void connectDevice(Device device) {
+    if (!isADBInstalled()) {
+      view.showADBNotInstalledNotification();
+      return;
+    }
+
+    List<Device> connectedDevices = new ArrayList<Device>();
+    connectedDevices.add(device);
+    connectedDevices = adb.connectDevices(connectedDevices);
+    for (Device connected : connectedDevices) {
+      updateDeviceInfo(connected);
+    }
+    showConnectionResultNotification(connectedDevices);
+  }
+
+  public void disconnectDevice(Device device) {
+    if (!isADBInstalled()) {
+      view.showADBNotInstalledNotification();
+      return;
+    }
+
+    List<Device> disconnected = new ArrayList<Device>();
+    disconnected.add(device);
+    disconnected = adb.disconnectDevices(disconnected);
+    for (Device d : disconnected) {
+      updateDeviceInfo(d);
+    }
+    showDisconnectionResultNotification(disconnected);
+  }
+
+  private void showConnectionResultNotification(List<Device> devices) {
+    for (Device d : devices) {
+      if (d.isConnected()) {
+        view.showConnectedDeviceNotification(d);
+      } else {
+        view.showErrorConnectingDeviceNotification(d);
+      }
+    }
+  }
+
+  private void showDisconnectionResultNotification(List<Device> devices) {
+    for (Device d : devices) {
```

Comment:
It's ok to use really short variable names for the array objects, local variable names, etc.
But, it's up to you, fixed ;)